torchprime/torch_xla_models/model/base_causal_lm.py

```diff
@@ -100,8 +100,10 @@ def export(self, save_directory: str):
     with open(os.path.join(save_directory, "config.json"), "w") as f:
       json.dump(OmegaConf.to_container(self.config, resolve=True), f, indent=2)
 
-  def from_pretrained(self, model_path_or_repo: str):
-    """Load model weights from local directory or Hugging Face Hub repo.
+  def from_pretrained(
```

Comment:
it doesn't look like a right API design to have from_pretrained to take save_directory and doing some saving.

There are 2 ways:
- from_pretraining to store some metadata internally about config and tokenizer and saves it along with other files in save_checkpoint (guarded by additional argument)
- Add new save_huggingface_configs method that downloads and saves configs

Reply:
let me move saving config to a separate save_huggingface_configs method